### ISSUE TITLE
Compilation with -D display ?

### ISSUE THREAD

**Issue by ncannasse** [vshaxe/vshaxe#245]
This can be either an error on VSHaxe or a context mismatch in Haxe compilation server.
When refactoring some code I managed to get the following two errors:

`haxe/macro/PositionTools.hx:71: Class<haxe.macro.Context> has no field load`
`haxe/std/haxe/macro/TypedExprTools.hx:161: Class<haxe.macro.Context> has no field sExpr`

Both are actually compiled with `#if (macro || display)` which makes me think that display is defined when I compile. I know that toggling display does not change the compiler context (because that would never use the cache) but it shouldn't be defined with compiling, so not sure what's going on here.

This error have occurred a few times in the past days.

**Comment by Simn** [comment]
Oh dear, maybe we resurrected some instance of https://haxe.org/blog/how-to-forge-signatures-using-a-compiler/

**Comment by ncannasse** [comment]
I don't think so. Compilation with or without -D display should have the same signature.

**Comment by Gama11** [comment]
I don't see how this could be a problem on the vshaxe side, it doesn't add `-D display` on compilation requests. You can see the exact arguments with this:

```json
"haxe.displayServer": {
    "arguments": ["-v"],
},
```

**Comment by Gama11** [comment]
@dmitryhryppa That image seems to be broken, what was it meant to show?

**Comment by back2dos** [comment]
FYI: you can paste / drop images into GitHub issues directly ;)

**Comment by dmitryhryppa** [comment]
Sorry for the late reply.
Here, I tried to reproduce what was on the screenshot. 
When I typing any code, the macro function is not looking into the `#display` define:

![screen shot 2019-02-11 at 17 36 58](https://user-images.githubusercontent.com/4466595/52573864-b6602c80-2e23-11e9-9556-2c0f77230b04.png)

**Comment by Gama11** [comment]
@dmitryhryppa I'm actually seeing the following behavior:

- Toplevel completion (just hit <kbd>Ctrl</kbd>+<kbd>Space</kbd> in an empty spot) triggers the `#if display` case in `test()`.
- Field completion like in your example doesn't re-run `test()` _at all_.
- The traces you were seeing there were likely from diagnostics (triggered on save) - use `"haxe.enableDiagnostics": false` to make it easier to test that. I think `display` not being defined there is the expected behavior.

Also, this seems unrelated to the original issue here, since it's not related to compilation?

**Comment by dmitryhryppa** [comment]
Ah, got it. That was weird for me because when I used FD then`!display` guarantee me that some macro code will not be invoked while I'm not compiling the project myself. Seems this is not true for VSCode, because it is doing "hidden" compilations, yes?

> Also, this seems unrelated to the original issue here, since it's not related to compilation?

I did not know what is going on, so that's why I asked it here to prevent duplicates :) Sorry.  Feel free to delete my posts.
